src/Microsoft.Extensions.CodeGeneration.Core/DefaultCodeGeneratorAssemblyProvider.cs

```diff
@@ -54,8 +51,12 @@ public IEnumerable<Assembly> CandidateAssemblies
                     .SelectMany(_libraryManager.GetReferencingLibraries)
                     .Distinct()
                     .Where(IsCandidateLibrary);
-                
+#if NET451
+                // For Net 451 the exe is being invoked from the build folder so all dependencies are next to the exe.
+                return list.Select(lib => _assemblyLoadContext.LoadFromName(new AssemblyName(lib.Identity.Name)));
```

Comment:
@NTaylorMullen having to build the project for net45 is necessary, but for core it is just going to add an overhead. 

Also, if there are build errors in the user's project, scaffolding can still work for adding non ef based controllers, which will also be affected in case the build is required.

Although it might be reasonable to fail to scaffold if there are build errors, scaffolding internally compiles the project (while scaffolding an ef based controller) in memory after modifying the project's startup and adding a DbContext (modifications also in memory) 
Requiring it to do build before invocation seems wasteful.